### PR TITLE
[Coop] Less efficient convert System.Diagnostics.{FileVersionInfo,Process}.

### PR DESCRIPTION
**Pull request by jaykrell**
This is inferior form of https://github.com/mono/mono/pull/15775 using preexisting slower forms.
This is really imho not the direction to go.

This code might not matter, but efficiency, micro and macro, should be favored esp. when it is easy as it is here, and efficient primitives provided. Accessing TLS and allocation coop handles should be minimized.

### CODE REVIEW

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
Looks ok, can't comment on the uwp changes.

**Comment by jaykrell** [comment]
https://github.com/mono/mono/pull/15776 going first would help, but let me take a quick stab at separating them here.

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
@monojenkins build failed

File: mono/metadata/w32process-win32-uwp.c [hunk 1]
```diff
@@ -9,205 +9,127 @@
...
-	gchar		*api_name = "";
-
-	if (mono_process_info->username) {
-		api_name = "CreateProcessWithLogonW";
-	} else {
-		api_name = "CreateProcess";
-	}
+	// FIXME CreateProcess is supported for UWP. Use finer grained #if.
 
-	memset (&process_info, 0, sizeof (PROCESS_INFORMATION));
```
Comment (jaykrell) [inline mono/metadata/w32process-win32-uwp.c]: Nice little memory corrupter here.

**Pull request merged**